jetcd-launcher/src/main/java/io/etcd/jetcd/launcher/EtcdContainer.java

```diff
@@ -102,12 +105,41 @@ public EtcdContainer withAdditionalArgs(Collection<String> additionalArgs) {
         return self();
     }
 
+    /**
+     * If `true` is passed as the parameter, start the Etcd Container as a non-root user.
+     * By default, {@link EtcdContainer#user} is "1000".
+     *
+     * @param rootless Refer to <a href="https://docs.docker.com/engine/reference/run/#user">User</a> .
+     * @return self container.
+     */
+    public EtcdContainer withRootless(boolean rootless) {
+        this.rootless = rootless;
+        return self();
+    }
+
+    /**
+     * {@link EtcdContainer#user} only makes sense if {@link EtcdContainer#rootless} is `true`.
+     * Optional values are {@code [ user | user:group | uid | uid:gid | user:gid | uid:group ]}.
+     * See <a href="https://docs.docker.com/engine/reference/run/#user">User</a> .
+     *
+     * @param user Refer to {@link com.github.dockerjava.api.command.CreateContainerCmd#withUser(String)}
+     * @return self container.
+     */
+    public EtcdContainer withUser(String user) {
+        this.user = user;
+        return self();
+    }
+
     @Override
     protected void configure() {
         if (!configured.compareAndSet(false, true)) {
             return;
         }
 
+        if (rootless && System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("linux")) {
+            withCreateContainerCmdModifier(cmd -> cmd.withUser(user));
```

Comment:
I think the only required check is against the fact that the user is set or not

Reply:
- In the new commmit I carried out another treatment.

Reply:
I think the check `System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("linux")` should be removed. If the developer set the user, then its choice should be honored.

Reply:
- Done.

Reply:
Note that the code already support setting a user with the env var` TC_USER` https://github.com/etcd-io/jetcd/blob/b51258e4a7c6587e16aacbcfbacf8901f8675878/jetcd-launcher/src/main/java/io/etcd/jetcd/launcher/EtcdContainer.java#L123-L126 soyour change should also take into account the env var.

Reply:
- I assume you mean that user settings for `user` should always take precedence over the value of the `TC_USER` environment variable? I made another change in the new commit.